spotify.go

```diff
@@ -124,6 +175,74 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+	retryDuration time.Duration
+}
+
+func (c *Client) ExecuteOpt(req *http.Request, needsStatus int, result interface{}) (err error) {
```

Comment:
- Does this need to be exported?
- What does `needsStatus` do?
- It seems like we should have a maximum number of retries to prevent getting stuck in an infinite loop.